src/components/ReusableTable/ReusableTable.component.tsx

```diff
@@ -0,0 +1,115 @@
+import { ReactNode } from 'react';
+
+import {
+    Table,
+    TableBody,
+    TableCell,
+    TableContainer,
+    TableHead,
+    TableRow,
+    Typography,
+    useMediaQuery,
+    useTheme,
+} from '@mui/material';
+
+import { formatDateVariants } from '@utils';
+
+import { HeadCellText, StatusChip } from './ReusableTable.styles';
+import type {
+    Category,
+    ReusableTableProps,
+    Status,
+} from './ReusableTable.types';
+
+// If you are not passing a category from one of the below, you have to pass custom formatter function
+// Category options: 'date', 'amount', 'status'
+// Example of custom formatter function:
+/*
+formatter: (row) => (
+    <Typography variant="body2" color="text.secondary">
+        {row.customField}
+    </Typography>
+)
+*/
+function renderCell<T>(category: Category | undefined, data: T): ReactNode {
```

Comment:
```suggestion
function renderCell<T>(category?: Category, data: T): ReactNode {
```

How is type defined here using T?